client/user/src/app/(nonRoot)/apply/_components/Step/TermsAgreement.tsx

```
@@ -1,9 +1,58 @@
+    }
+  }, [visibleTerms, setTerms])
+  if (visibleTerms.length === 0) return null
+
+  const isAllChecked = visibleTerms.every(
+    (term) => (terms as Record<number, boolean>)[term.id] || false
+  )
+
+  const handleAllAgreementChange = () => {
+    const newState = !isAllChecked
+    const updatedTerms = visibleTerms.reduce<Record<number, boolean>>(
+      (acc, term) => ({ ...acc, [term.id]: newState }),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)